src/test/java/uk/gov/hmcts/reform/sendletter/controllers/GetLetterStatusControllerTest.java

```diff
@@ -0,0 +1,77 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import uk.gov.hmcts.reform.authorisation.exceptions.InvalidTokenException;
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+import uk.gov.hmcts.reform.sendletter.domain.LetterStatus;
+import uk.gov.hmcts.reform.sendletter.exception.LetterNotFoundException;
+import uk.gov.hmcts.reform.sendletter.services.LetterService;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.never;
+import static org.mockito.BDDMockito.verify;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Matchers.anyString;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest
+public class GetLetterStatusControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private LetterService service;
+
+    @MockBean
+    private AuthTokenValidator tokenValidator;
+
+    @Test
+    public void should_return_letter_status_when_it_is_found_in_database() throws Exception {
+        given(tokenValidator.getServiceName("auth-header-value")).willReturn("service-name");
+        given(service.getStatus("message-id", "service-name")).willReturn(new LetterStatus("message-id"));
+
+        getLetter()
+            .andExpect(status().isOk())
+            .andExpect(content().json(
+                "{"
+                    + "\"message_id\":\"message-id\""
+                    + "}"
+            ));
+    }
+
+    @Test
+    public void should_return_404_client_error_when_letter_is_not_found_in_database() throws Exception {
+        given(tokenValidator.getServiceName("auth-header-value")).willReturn("service-name");
+        willThrow(LetterNotFoundException.class).given(service).getStatus("message-id", "service-name");
+
+        getLetter().andExpect(status().is(HttpStatus.NOT_FOUND.value()));
+    }
+
+    @Test
+    public void should_return_401_client_error_when_authorisation_header_is_invalid() throws Exception {
+        willThrow(InvalidTokenException.class).given(tokenValidator).getServiceName("auth-header-value");
+
+        getLetter().andExpect(status().is(HttpStatus.UNAUTHORIZED.value()));
+
+        verify(service, never()).getStatus(anyString(), anyString());
+    }
+
+    private ResultActions getLetter() throws Exception {
+        return mockMvc.perform(
+            get("/letters/message-id")
```

Comment:
this looks like a regular url at first glance, a sample uuid would be clearer imgo

Reply:
would `some-message-id` suffice?

Reply:
ok